File: elixir-resend-examples/examples/batch_send.exs
Mix.install([
  {:resend, "~> 0.4.5"},
  {:dotenvy, "~> 0.8"}
])

if File.exists?(".env"), do: Dotenvy.source!([".env"])

api_key = System.get_env("RESEND_API_KEY") || raise "RESEND_API_KEY environment variable is required"
from = System.get_env("EMAIL_FROM") || "Acme <onboarding@resend.dev>"
contact_email = System.get_env("CONTACT_EMAIL") || "delivered@resend.dev"

# Batch send: up to 100 emails per call
# Note: Batch send does not support attachments or scheduling
emails = [
  %{
    from: from,
    to: ["delivered@resend.dev"],
    subject: "We received your message",
    html: "<h1>Thanks for reaching out!</h1><p>We'll get back to you soon.</p>"
  },
  %{
    from: from,
    to: [contact_email],
    subject: "New contact form submission",
    html: "<h1>New message received</h1><p>From: delivered@resend.dev</p>"
  }
]

case Resend.Batch.send(emails) do
  {:ok, %{"data" => data}} ->
    IO.puts("Batch sent successfully!")

    data
    |> Enum.with_index(1)
    |> Enum.each(fn {email, index} ->
      IO.puts("Email #{index} ID: #{email["id"]}")
    end)

  {:error, reason} ->
    IO.puts(:stderr, "Error sending batch: #{inspect(reason)}")
    System.halt(1)
end

File: bun-resend-examples/javascript/examples/batch-send.js
import { Resend } from "resend";

const resend = new Resend(process.env.RESEND_API_KEY);

const from = process.env.EMAIL_FROM || "Acme <onboarding@resend.dev>";
const contactEmail = process.env.CONTACT_EMAIL || "delivered@resend.dev";

// Batch send: up to 100 emails per call
// Note: Batch send does not support attachments or scheduling
const { data, error } = await resend.batch.send([
  {
    from,
    to: ["delivered@resend.dev"],
    subject: "We received your message",
    html: "<h1>Thanks for reaching out!</h1><p>We'll get back to you soon.</p>",
  },
  {
    from,
    to: [contactEmail],
    subject: "New contact form submission",
    html: "<h1>New message received</h1><p>From: delivered@resend.dev</p>",
  },
]);

if (error) {
  console.error("Error sending batch:", error);
  process.exit(1);
}

console.log("Batch sent successfully!");
data?.data.forEach((email, i) => {
  console.log(`Email ${i + 1} ID: ${email.id}`);
});
